src/rpc/format/mod.rs

```diff
@@ -0,0 +1,74 @@
+mod bincode;
+pub mod cbor;
+mod json;
+pub mod msgpack;
+mod revision;
+
+use crate::rpc::failure::Failure;
+use crate::rpc::request::Request;
+use crate::rpc::response::Response;
+use axum::extract::ws::Message;
+
+pub const PROTOCOLS: [&str; 5] = [
+	"json",     // For basic JSON serialisation
+	"cbor",     // For basic CBOR serialisation
+	"msgpack",  // For basic Msgpack serialisation
+	"bincode",  // For full internal serialisation
+	"revision", // For full versioned serialisation
+];
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Format {
+	None,     // No format is specified yet
+	Json,     // For basic JSON serialisation
+	Cbor,     // For basic CBOR serialisation
+	Msgpack,  // For basic Msgpack serialisation
+	Bincode,  // For full internal serialisation
+	Revision, // For full versioned serialisation
+}
+
+impl From<&str> for Format {
+	fn from(v: &str) -> Self {
+		match v {
+			s if s == PROTOCOLS[0] => Format::Json,
+			s if s == PROTOCOLS[1] => Format::Cbor,
+			s if s == PROTOCOLS[2] => Format::Msgpack,
+			s if s == PROTOCOLS[3] => Format::Bincode,
+			s if s == PROTOCOLS[4] => Format::Revision,
+			_ => Format::None,
+		}
+	}
+}
+
+impl Format {
+	/// Check if this format has been set
+	pub fn is_none(&self) -> bool {
+		matches!(self, Format::None)
+	}
+	/// Check if strings in this format might be Record IDs
+	pub fn check_strings(&self) -> bool {
+		matches!(self, Format::Cbor | Format::Msgpack)
```

Comment:
The JSON format currently also checks strings, this will be a breaking change if I'm not mistaken

Reply:
The SurrealQL (JSON) format, parses the whole query as SurrealQL, and therefore UUIDs, Record IDs, Datetimes and other types are already in the correct format. This is a quick workaround until we make use of tags in CBOR and MessagePack - once we implement tags in CBOR and MessagePack, then we would expect the special SurrealQL types to be tagged correctly, and not as strings.

Reply:
Gotcha! If not much work, should we rework the JSON syntax parser to accept an option to check strings? From what I remember it of the JSON syntax parser this should not be much works and makes expectations more clear